### DISCUSSION TITLE
v1.0.0-beta.10

### DISCUSSION THREAD

**Discussion by Taureon** [Taureon/aps-plus-plus#89] [General]
* Added a new System for Split Upgrades (like in Scenexe):
  - Tanks and upgrades can be defined as an array of strings.
  - Every definition contributes its guns, body stats, turrets, upgrades, and other things.
  - Each definition can be upgraded separately. (e.g. Assassin-Twin  -> Assassin-Bulwark)
  - Upgrades can be batched with `BATCH_UPGRADES`, listing every combination and forcing all branches to be upgraded at the same time.
  - Upgrade branches can have labels with `BRANCH_LABEL`.
  - Client-side upgrade limit raised from 32 to 100.

* Added a way better upgrade tree.
  - Move in all 4 directions with arrow keys.
  - Hold `SHIFT` to move faster.
  - Support for theoretically infinite tiers.
  - Visually upgraded.
  - Has tank labels.

* Updated the Chatbox.
  - Added the option to press `ESC` to close the chat input.
  - Added the ability to close the chat input with `ENTER` if you didn't write any text.

* Added `GPL-3.0` as the license.

* Added upgrade paths for `healer` tank.

* Added `UPGRADE_LABEL` entity attribute, which overrides the `LABEL` for the upgrade picker.

* Added load time statistics for definitions loading, addon loading and mockup generation.

* Added a color cache for `modifyColor`.

* Added more statnames.

* Reworked the dev menu.

* Moved generic client-side color-related functions to gameDraw.js.

* Made a better fix for adding `healer` tank to `basic`'s upgrade list if the gamemode is siege.

* Made Sphere shine.

* Improved client-side turret rendering.

* Improved FOV to be similar to how arras.io's FOV changes as you grow.

* Fixed complex food (Relics and Laby shapes) shapeshifting when spinning.

* Fixed healer-tanks' stat names.

* Fixed an error related to controllers resetting.

* Fixed an error related to `combineStats`' backwards compatibility.

* Fixed more mistakes that don't really matter.

* Fixed a Manhunt related crash.

* Fixed a crash related to `modifyColor`.

* Minor default-configuration changes.

<hr /><em>This discussion was created from the release <a href='https://github.com/Taureon/aps-plus-plus/releases/tag/v1.0.0-beta.10'>v1.0.0-beta.10</a>.</em>

**Comment by d3ltastorm** [comment]
o
